spec/models/user_spec.rb

```diff
@@ -12,6 +12,16 @@
       expect(FactoryGirl.build(:user, :email => "thisguy@@manageiq.com")).not_to be_valid
     end
 
+    it "should invalidate email addresses with uppercase letters" do
```

Comment:
**tl; dr;**  I don't really feel strongly about needing to keep this here, but that below is the reason why I think it should be there:

I see this test in `more_core_extensions` testing that the regexp doesn't match a string when capital letters are present, and this test here is confirming that the `User` model fails when an email is present with capital letters.  I see this as a subtle, but important difference.

The difference I see is that one is testing that the lib is doing it's job, and the other is testing the application behaves as we expect.  Part of what this PR is doing is making a concrete declaration that emails for `Users` are required to be lower case, regardless of what underlying mechanics are used under the hood to do that.  This means that if we decided to stop using `more_core_extensions` (fat chance of that happening... but still), this test would be there to catch us when that changes.

Also, you made this comment:

> I think relaxing the email validation might bring up new problems - then we have to make sure we downcase everywhere we use an email.

Which really makes two statements:

a)  We think that emails are required to be downcased, but we are not sure what the full extent of relaxing those conditions would be.
b)  The other edge case with LDAP (links in the description), are clearly not being tested against in our application because this hasn't been a problem via the test suite.

This test basically solves those two by adding a test that confirms that those two are tested against to some degree (less so on `b` though).

To your point, this test ends up being some what redundant, but it is quick and not that much code that I won't loose sleep over it.  Also, I see `more_core_extensions` as a library that we are testing it's integration of with our app, versus something like `manageiq-providers-amazon` which is a component of our application that is in a separately contained repo.

That said, if you are talking about the second test here, I would agree with you that it is pretty garbage and should probably be thrown out... but you know... _**TDD and Whatnot™**_